docs/core/knowledge-bases.rst

```diff
@@ -0,0 +1,383 @@
+Other requests, such as comparison between objects, are currently not supported.
+
+Another thing you may have noticed is, that we marked different kind of entities in the NLU data.
+If you want to use ``ActionQueryKnowledgeBase``, you need to specify the following entities:
+
+- ``object_type``: Whenever the user is talking about a specific object type from your knowledge base, the type should
+  be marked as entity in our NLU data. Use :ref:`entity_synonyms` to map, for example, "restaurants" to the correct
+  object type listed in the knowledge base, e.g. "restaurant".
+- ``mention``: If the user refers to an object via "the first one", "that one", or "it", you should mark those terms
+  as ``mention``. We also use :ref:`entity_synonyms` to map some of the mentions to symbols. More on that in
+  :ref:`resolve_mentions`.
+- ``attribute``: All attribute names defined in your knowledge base should be marked as ``attribute`` in the NLU data.
+  Again, use :ref:`entity_synonyms` to map variations of an attribute name to the one used in the knowledge base.
+
+Don't forget to add those entities to your domain file (as entities and slots).
+
+Query the Knowledge Base for Objects
+------------------------------------
+
+In order to query the knowledge base for any kind of objects, the user's request needs to include the object type.
+Otherwise, the action does not know what objects the user is interested in and cannot formulate the query.
+
+The user may restrict his request to a specific kind of object.
+For example, he could say ``What Italian restaurant options in Berlin do I have?``.
+In this example the user want to obtain a list of restaurants that (1) have an Italian cuisine and (2) are located in
+Berlin.
+In order to filter the objects in the knowledge base, you need to mark "Italian" and "Berlin" as entities.
+E.g. ``What [Italian](cuisine) [restaurant](object_type) options in [Berlin](city) do I have?``.
+The names of the attributes, e.g. "cuisine" and "city", should be equal to the ones used in the knowledge base.
+You also need to add those entities as entities and slots in the domain file.
+If the NER detects those attributes in the request of the user, the action will use those for filter the
+restaurants found in the knowledge base.
+
+Once the bot retrieved some entities from the knowledge base, it will response to the user with
+
+    `Found the following objects of type 'restaurant':`
+    `1: I due forni`
+    `2: PastaBar`
+    `3: Berlin Burrito Company`
+
+Or if no entities could be found
+
+    `I could not find any objects of type 'restaurant'.`
+
+If you want to change the utterance of the bot, you can overwrite the method ``utter_objects()`` in your action.
+
+Query the Knowledge Base for an Attribute of an Object
+------------------------------------------------------
+
+To obtain the value of an attribute for a specific object from the knowledge base, the action needs to know the object
```

Comment:
This is super difficult language: "To obtain the value of an attribute for a specific object" - I think we can say this in plain english by usign an example